### ISSUE TITLE
URL::site() should always rawurlencode uri

### ISSUE THREAD

**Issue by kohamkohane** [kohana/core#631]
Now URL::site() use rawurlencode only if UTF8::is_ascii($path) is FALSE. Why? I suggest always encode uri. For example space is ascii and should be encoded!

**Comment by enov** [comment]
Yeah, I think you are right on this one. I did not do any tests, but even those URL delimiters `?` and `&` are ASCII. Could you provide a PR to fix with unit tests (if applicable)?

**Comment by kohamkohane** [comment]
I could, but first let's think how rawurlencode should works in kohana. If we add rawurlencode to URL:site (in my opinion the best choicie) then we should not rawurlencode in route->url() becouse it use URL:site() so we double rawurlencode then. I think we should only rawurlencode in URL::site() no in route. What do you think about it?

By the way, does route->uri() (URI not URL) should be rawurlencoded too?

**Comment by kohamkohane** [comment]
Or maybe different view. Rawurlencode parameters in route->uri() as now, and do not rawurlencode in URL:site() or rawurlencode only if URI passed to URL:site() was not encoded earilier. 

Anyway current if(UTF8::is_ascii($path)) in URL::site is wrong.

**Comment by enov** [comment]
I am leaning towards removing `rawurlencode` from `URL::site` as it is assuming that a forward slash should always be treated as URL parameter separator.

There is no way to encode a slash in there, even if the slash itself is a sub-string of a parameter. `URL::site`, unlike `Route::uri`, does not have prior knowledge of parameters.

But that means breaking existing applications, so it should be done in v3.4.

Let me know what do you think.

**Comment by enov** [comment]
cc @acoulton

**Comment by kohamkohane** [comment]
I agree with you. Just remove rawurlencode from URL::site.

**Comment by acoulton** [comment]
@enov I think I agree - remove rawurlencode from URL::site
